### ISSUE TITLE
Positioning the Socialite url bar icon fails if the feed icon is not part of the location bar.

### ISSUE THREAD

**Issue by chromakode** [reddit-archive/socialite#1]
It also doesn't work for me when Firefox Subscribe button is displayed on any toolbar. Error Console reports:

```
Error: uncaught exception: [Exception... "Node was not found"
code: "8" nsresult: "0x80530008 (NS_ERROR_DOM_NOT_FOUND_ERR)"
location: "chrome://socialite/content/siteUrlBarIcon.js Line: 89"]
```

Problem is that you still use Subscribe button (feed-button) as positioning method for location bar icon but in Firefox 4 it's standalone button and no longer part of location bar.

see http://www.reddit.com/r/socialite/comments/h7rjs/socialite_14_firefox_4_please_test/c1twshr?context=3

**Comment by nikola-kocic** [comment]
I made small edits that fixed this issue, you can find file here: http://dl.dropbox.com/u/37621389/socialite_no_feed_bug.xpi
In short, I replaced 

```
urlBarIconParent.insertBefore(urlBarIcon, feedButton);
```

with

```
urlBarIconParent.appendChild(urlBarIcon);
```

and <code>nativeJSON.decode / nativeJSON.encode</code> with <code>JSON.parse / JSON.stringify</code> because they are deprecated in Firefox 7 (and are supported since Firefox 3.5)
-   Modified files: 
  
  > \modules\site.jsm
  > \modules\reddit\redditAPI.jsm
  > \modules\reddit\authentication.jsm
  > \content\siteUrlBarIcon.js

**Comment by chromakode** [comment]
Thanks! Will review and merge up. Are you able to make a pull request with your changes?

**Comment by nikola-kocic** [comment]
Sorry, I don't know how to work with github. You do whatever you need to do, no need to credit me or anything.

**Comment by chromakode** [comment]
I would prefer to credit the commit to you, since it's your work. Making a pull request is well documented and easy enough to learn, and it won't be the last time it comes in handy! http://help.github.com/send-pull-requests/

Alternatively, could you provide me with a name+email so I can attribute the commit to you?

**Comment by chromakode** [comment]
Merged up the fix in ec884f7d54e6caa68b0d4a53849575168502db87
